From 3fcebb9b38b8b01e528d158656af863f98a3c751 Mon Sep 17 00:00:00 2001
From: Simon Josefsson <simon@josefsson.org>
Date: Fri, 3 Sep 2021 15:59:45 +0200
Subject: [PATCH] telnet: Don't infloop for malicious server.

See https://bugs.debian.org/945861 and the tiny patch used by NetBSD:
https://github.com/NetBSD/src/commit/36b8cfb2e28f691beae12da0c207086e1df0c8c4#diff-38b7213f9c6c21245fbeb4fad9520a27239d712a6dd0fea20dd6b77203b5737c

* telnet/utilities.c (ExitString): Don't call SetForExit().

diff --git a/telnet/utilities.c b/telnet/utilities.c
index 85fd4762..08571cd4 100644
--- a/telnet/utilities.c
+++ b/telnet/utilities.c
@@ -1063,7 +1063,6 @@ Exit (int returnCode)
 void
 ExitString (char *string, int returnCode)
 {
-  SetForExit ();
   fwrite (string, 1, strlen (string), stderr);
   exit (returnCode);
 }
-- 
2.33.0

